shared/runtime/sys_stdio_mphal.c

```diff
@@ -54,7 +54,7 @@ static const sys_stdio_obj_t stdio_buffer_obj;
 
 static void stdio_obj_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     sys_stdio_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    mp_printf(print, "<io.FileIO %d>", self->fd);
+    mp_printf(print, "<io.StringIO %d>", self->fd);
```

Comment:
This will now print the wrong value for things like `sys.stdin.buffer`, which should remain a `FileIO`.

Could fix that with the following:
```c
mp_printf(print, "<io.%s %d>", mp_obj_get_type_str(self_in), self->fd);
```